### ISSUE TITLE
Display acronym instead of name for datasets

### ISSUE THREAD

**Issue by ocrasborn** [Signbank/Global-signbank#541]
We added the field 'acronym' to the metadata for each dataset (like on https://signbank.science.ru.nl/datasets/detail/5), but we still display the full name everywhere. Please change that:
* In the menu bar where the list of datasets appears
* In the list view of search results
* In the detail view
* Any other location that I forget now?
I've changed the name of NGT to Nederlandse Gebarentaal (Acronym NGT), so that it will be clearer where it has yet to be done.

**Comment by Woseseltops** [comment]
Some things that came out of a conversation with @susanodd 
* Often there is no acronym, so we need some kind of function that checks if the acronym is there and display the normal name if not. I already have a function to create a short name (for the menu bar), so we just have to revise and rename that and then start using it at other places as well.
* We noticed you renamed the NGT dataset @ocrasborn . This might give problems when reimporting exports created before you did this.
* Related: do we also want to show the acronym in exports? Because that again will cause problem when importing old exports.

**Comment by ocrasborn** [comment]
Ah, I see. The background to this wish: for many SLs like NGT, everyone is familiar with the acronym and uses that as the name. For others, this is less the case, so we decided to label the field 'Name' initially. Now in second instance, it seems to be a good idea to separate the acronym/short name from the full name for some sign languages. So I'd like to shift to using the acronym throughout, and only use the Name on the dataset metadata page.
* I'll fill in an acronym everywhere now. Could you make it obligatory when creating a new dataset from now on?
* Your comment on the import function makes me realise that there will be more places where we need to move from Name to Acronym in relation to datasets. Shall we put that in a separate issue (export-import)?
* Users rarely if ever re-use old export files for importing, and if they did, they can easily rename the field in the CSV. So let's not bother about that. But is it a problem for the developers elsewhere perhaps, in reusing old migration files?

**Comment by vanlummelhuizen** [comment]
@ocrasborn Another issue when naming a Dataset is that Dataset names are used when generating an ECV, for both the file name (which is included in the url) and the CV name, which is inside the ECV. Since an ECV might be used in EAFs, changing the Dataset name also involves changing stuff in EAFs. 

A separate ECV name field might be a solution, but that may also clutter the Dataset naming functionality. Any other ideas?

**Comment by susanodd** [comment]
Using acronyms also affects the tests. They are all failing now because they look for the Dataset named NGT. I'm working on the tests file because of CSV, so I'm fixing this too.

**Comment by ocrasborn** [comment]
Feel free to rename the dataset back to NGT for the moment if that makes your work easier, @susanodd and @vanlummelhuizen !

**Comment by vanlummelhuizen** [comment]
@ocrasborn @susanodd Please do not do that! I have changed all references in the CNGT EAFs to deal with the new name and changing it back will break it again!

**Comment by ocrasborn** [comment]
:-)

**Comment by ocrasborn** [comment]
@susanodd : better base the tests on the acronym, right, rather than altering the name.

**Comment by susanodd** [comment]
It's not as easy as that. That's what I tried first. But the tests also call "forms" which then in turn also use the name field. So these forms also need to be revised.
In order to solve #535 , I have commented out some of the other tests that also seem to be broken and just focus on the test mentioned in that issue.

**Comment by Woseseltops** [comment]
There's a bit of confusion about this issue (at least on my side). 

*What I still understand*: @susanodd changed the code on various places so it uses dataset.acronym instead of dataset.name to identify datasets. As a result of this, it doesn't matter that the DEFAULT_DATASET setting is still NGT (instead of Nederlandse Gebarentaal).

*What confuses me*: to get the code running on Susan's local machine, we need to change the DEFAULT DATASET setting to 'Nederlandse Gebarentaal', but this is not needed for the live server.

*What worries me*: assuming that DEFAULT_DATASET expects an acronym... how can other people know this? Wouldn't it be better rename this setting DEFAULT_DATASET_ACRONYM?

**Comment by susanodd** [comment]
The tests also make use of acronyms.

**Comment by susanodd** [comment]
A setting for acronym has been added and the code modified to reflect this.

**Comment by vanlummelhuizen** [comment]
Is this all done @susanodd and @Woseseltops ?

**Comment by susanodd** [comment]
Yes. There's one spot I noticed when looking at the ECV files. It's not clear whether the acronym should be used or the dataset name. The problem is "going in reverse". The acronyms don't have spaces in them. But the dataset name, the space is converted to "_". So the names need to be mapped between "what the user defined" (e.g., with mix of small and cap letters), and whatever filename is in the ecv folder.
(There is now a test to check that the ecv files have glosses - #563. That has been implemented. But the filenames on disc are/needed to be reverse mapped to the dataset. There were some files that didn't seem to belong to any dataset anymore.)
